src/ClientNet462/Program.cs

```diff
@@ -0,0 +1,148 @@
+                        var result = await CallGraphQLAsync<MyResponseData>(
+                            new Uri("https://graphql-weather-api.herokuapp.com/"),
+                            HttpMethod.Post,
+                            "query ($city: String!) { getCityByName(name: $city) { name country } }",
+                            new
+                            {
+                                city = "detroit",
+                            },
+                            cts.Token);
+                        if (result.Errors?.Count > 0)
+                        {
+                            Console.WriteLine($"GraphQL returned errors:\n{string.Join("\n", result.Errors.Select(x => $"  - {x.Message}"))}");
+                        }
+                        else
+                        {
+                            Console.WriteLine($"Found city {result.Data.GetCityByName.Name}, {result.Data.GetCityByName.Country}");
+                        }
+                    }
+                    finally
+                    {
+                        Console.CancelKeyPress -= handler;
+                    }
+                }
+            }
+            catch (OperationCanceledException)
+            {
+                Console.WriteLine("Cancelled query");
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Caught exception: {ex.Message}");
+            }
+            Console.WriteLine("Press enter to exit");
+            Console.ReadLine();
+        }
+
+        private class MyResponseData
+        {
+            public MyResponseGetCityByName GetCityByName { get; set; }
+        }
+
+        private class MyResponseGetCityByName
+        {
+            public string Name { get; set; }
+            public string Country { get; set; }
+        }
+
+        private static HttpClient httpClient = new HttpClient();
+
+        static async Task<GraphQLResponse<TResponse>> CallGraphQLAsync<TResponse>(Uri endpoint, HttpMethod method, string query, object variables, CancellationToken cancellationToken)
+        {
+            var content = new StringContent(SerializeGraphQLCall(query, variables), Encoding.UTF8, "application/json");
+            var httpRequestMessage = new HttpRequestMessage
+            {
+                Method = method,
+                Content = content,
+                RequestUri = endpoint,
+            };
+            //add authorization headers if necessary here
+            httpRequestMessage.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            using (var response = await httpClient.SendAsync(httpRequestMessage, cancellationToken))
+            {
+                //if (response.IsSuccessStatusCode)
+                if (response?.Content.Headers.ContentType?.MediaType == "application/json")
+                {
+                    var responseString = await response.Content.ReadAsStringAsync(); //cancellationToken supported for .NET 5/6
```

Comment:
Don't know why, but none of the methods on `HttpContent` prior to .NET 5 support a cancellation token here.  However, the sample still works because most of the waiting period is during `SendAsync`, which properly supports the cancellation token.

Reply:
Does not matter